fdbrpc/AsyncFileNonDurable.actor.h

```diff
@@ -267,6 +267,10 @@ class AsyncFileNonDurable final : public IAsyncFile, public ReferenceCounted<Asy
 			Future<Void> deleteFuture = deleteFile(this);
 			if (!deleteFuture.isReady())
 				filesBeingDeleted[filename] = deleteFuture;
+		} else if (isSoleOwner()) {
+			// isSoleOwner is a bit confusing here. What we mean is that the openFiles map is the sole owner. If we
+			// remove the file from the map to make sure it gets closed.
+			g_simulator.getCurrentProcess()->machine->openFiles.erase(filename);
```

Comment:
I was told once upon a time (I think by @sfc-gh-satherton) that not closing Files was intentional, as it then simulation tests the situation where we write to a file, close it, later (re-)open it, and can read the written data, but that written data isn't durable on disk.

When I had looked at this before, I sketched out something in #2766 that would close file descriptors but not Files.  I don't have the memory of the code anymore though to actually remember if this code actually does reproduce/test the situation I described above though.